packages/@aws-cdk/aws-servicecatalogappregistry/lib/private/validation.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Class to validate that inputs match requirements.
+ */
+export class InputValidator {
+  /**
+   * Validates length is between allowed min and max lengths.
+   */
+  public static validateLength(resourceName: string, inputName: string, minLength: number, maxLength: number, inputString?: string): void {
+    if (inputString !== undefined && (inputString.length < minLength || inputString.length > maxLength)) {
+      throw new Error(`Invalid ${inputName} for resource ${resourceName}, must have length between ${minLength} and ${maxLength}, got: '${this.truncateString(inputString, 100)}'`);
+    }
+  }
+
+  /**
+   * Validates a regex.
+   */
+  public static validateRegex(resourceName: string, inputName: string, regex: RegExp, inputString?: string): void {
+    if (inputString !== undefined && !regex.test(inputString)) {
```

Comment:
Same comment here as in `validateLength()` regarding dynamic CloudFormation expressions.